build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts

```diff
@@ -38,33 +38,18 @@ plugins {
 version =
     providers.fileContents(rootProject.layout.projectDirectory.versionTxt()).asText.get().trim()
 
-// Fail the build if Gradle is started with an old version of Java
-val MIN_JAVA_VERSION = JavaVersion.VERSION_21
-val CUR_JAVA_VERSION = JavaVersion.current()
-
-if (CUR_JAVA_VERSION.ordinal < MIN_JAVA_VERSION.ordinal) {
-    throw StopExecutionException(
-        "ERROR: Gradle is started with Java " +
-            CUR_JAVA_VERSION +
-            ". This project requires running Gradle with Java " +
-            MIN_JAVA_VERSION +
-            " or above." +
-            " Please check your JAVA_HOME and/or PATH and configure the default JDK to use Java version " +
-            MIN_JAVA_VERSION +
-            " or above."
+// Fail the build if Gradle is started with a Java version that does not exactly match
+val javaVersion = "21.0.1"
+val currentJavaVersion = providers.systemProperty("java.version").get()
+
+if (currentJavaVersion != javaVersion) {
+    throw RuntimeException(
+        "Gradle runs with Java $currentJavaVersion. This project requires Gradle to run with Java $javaVersion. " +
+            "\n - From commandline: change JAVA_HOME and/or PATH to point at Java $javaVersion installation." +
+            "\n - From IntelliJ: change 'Gradle JVM' in 'Gradle Settings' to point at Java $javaVersion installation."
     )
 }
 
-java {
-    sourceCompatibility = MIN_JAVA_VERSION
-    targetCompatibility = MIN_JAVA_VERSION
-
-    toolchain {
-        languageVersion = JavaLanguageVersion.of(21)
-        vendor = JvmVendorSpec.ADOPTIUM
-    }
-}
-
```

Comment:
This does not add anything anymore, as Gradle will use the JDK it is running with by default. And we now check before that that is the one we need for everything.